proposals/lip-0045.md

```diff
@@ -984,166 +942,357 @@ The Interoperability module contains the following commands:
-            return False
+##### Data
 
-    return terminateChainInternal(chainID)
+```java
+ccmProcessedDataSchema = {
+    "type": "object",
+    "required": ["ccmID", "result"],
+    "properties": {
+        "ccmID": {
+            "dataType": "bytes",
+            "length": HASH_LENGTH, 
+            "fieldNumber": 1
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
 ```
 
-#### getChainAccount
+#### TerminatedStateAccountCreated
 
-The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_STATE_CREATED`. This event is emitted when a new entry is created in the terminated state substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated chain account.
 
-##### Returns
+##### Data
 
-This function returns `chainAccount(chainID)`.
+The data property follows the `terminatedStateAccountSchema` schema given [above](#json-schema-5).
 
-#### getChannel
+#### TerminatedOutboxCreated
 
-The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_OUTBOX_CREATED`. This event is emitted when a new entry is created in the terminated outbox substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated outbox account.
 
-##### Returns
+##### Data
 
-This function returns `channel(chainID)`.
+The data property follows the `terminatedOutboxAccountSchema` schema given [above](#json-schema-6).
 
+### Protocol Logic for Other Modules
 
-#### getOwnChainAccount
+#### send
 
-The `getOwnChainAccount` function returns the object stored in the own chain data substore.
+The `send` function is used to create and add a message to the outbox of a partner chain.
 
-##### Parameters
+```python
+def send(sendingAddress: Address, 
+    module: Module, 
+    crossChainCommand: CrossChainCommand, 
+    receivingChainID: ChainID, 
+    fee: uint64, 
+    status: uint32, 
+    params: object
+) -> None:
+# Create cross-chain message
+    paramsSchema = JSON schema associated with (module, crossChainCommand)
+    ccm = {
+        "nonce": ownChainAccount.nonce,
+        "module": module,
+        "crossChainCommand": crossChainCommand,
+        "sendingChainID": ownChainAccount.chainID,
+        "receivingChainID": receivingChainID,
+        "fee": fee,
+        "status": status,
+        "params": encode(paramsSchema, params)
+    }
 
-This function takes no input.
+    # Validate ccm size
+    try:
+        validateFormat(ccm)
+    except Exception as e:
+        emitPersistentEvent(
+            module = MODULE_NAME_INTEROPERABILITY,
+            name = EVENT_NAME_CCM_SENT,
+            data = {"ccmID": ccmID, "result": ERROR_CODE_INVALID_FORMAT},
```

Comment:
`ccmID` is only computed below. I am not sure you can even compute a CCM ID if the format is invalid.

Reply:
Checking the `validateFormat` function, the encoding should always work and actually only the message might be too large or the module/command name may have invalid format.

Reply:
fixed